GaudiPluginService/src/PluginServiceV2.cpp

```diff
@@ -140,80 +143,70 @@ namespace Gaudi {
         void Registry::initialize() {
           REG_SCOPE_LOCK
 #if defined( _WIN32 )
-          const char* envVar = "PATH";
-          const char  sep    = ';';
+          const std::string envVar = "PATH";
+          const std::string sep    = ";";
 #elif defined( __APPLE__ )
-          const char* envVar = "DYLD_LIBRARY_PATH";
-          const char  sep    = ':';
+          const std::string envVar = "DYLD_LIBRARY_PATH";
+          const std::string sep    = ":";
 #else
-          const char* envVar = "LD_LIBRARY_PATH";
-          const char  sep    = ':';
+          const std::string envVar = "LD_LIBRARY_PATH";
+          const std::string sep    = ":";
 #endif
 
           std::regex  line_format{"^(?:[[:space:]]*(?:(v[0-9]+)::)?([^:]+):(.*[^[:space:]]))?[[:space:]]*(?:#.*)?$"};
-          std::smatch m;
+          std::smatch matches;
 
-          std::string search_path = std::getenv( envVar );
-          if ( !search_path.empty() ) {
-            logger().debug( std::string( "searching factories in " ) + envVar );
+          std::string search_path = std::getenv( envVar.c_str() );
+          if ( search_path.empty() ) {
+            return;
+          }
 
-            std::string::size_type start_pos = 0, end_pos = 0;
-            while ( start_pos != std::string::npos ) {
-              // correctly handle begin of string or path separator
-              if ( start_pos ) ++start_pos;
+          logger().debug("searching factories in " + envVar);
+          logger().debug("searching factories in " + search_path);
 
-              end_pos = search_path.find( sep, start_pos );
-              if ( end_pos == start_pos )   {
-                start_pos = std::string::npos;
-                continue;
-              }
-              fs::path dirName =
-#ifdef USE_BOOST_FILESYSTEM
-                  std::string{search_path.substr( start_pos, end_pos - start_pos )};
-#else
-                  search_path.substr( start_pos, end_pos - start_pos );
-#endif
-              start_pos = end_pos;
+          std::vector<std::string> directories;
+          boost::split(directories, search_path, boost::is_any_of(sep));
```

Comment:
Wouldn't it be maybe better to use `strtok` then if you build with C++17 it's independent of boost.

Reply:
But boost is anyway needed for other things.
And it is definitely more readable, and doesn't need one to deal with `char*`
```diff
-          std::vector<std::string> directories;
-          boost::split(directories, search_path, boost::is_any_of(sep));
-
-          for(fs::path dirName: directories) {
+          std::vector<char> cSP(search_path.c_str(), search_path.c_str() + search_path.size() + 1);
+          char* c_dirName = std::strtok(&cSP[0], sep.c_str());
+          while(c_dirName != nullptr) {
+            fs::path dirName(std::string{c_dirName});
+            c_dirName = std::strtok(nullptr, sep.c_str());

```